src/angular-masonry.js

```diff
@@ -65,7 +65,10 @@
             // Keep track of added elements.
             bricks[id] = true;
             defaultLoaded(element);
-            $element.masonry(method, element, true);
+            // Don't add element to Masonry if it already has it.
+            if ($element.masonry('getItemElements').indexOf(element.get(0)) === -1) {
```

Comment:
This would be O(n^2) when adding a list of n new elements, this seems like a very dangerous default. Is there any other way to address this?